src/Raven.Studio/typescript/models/database/documents/timeSeriesPolicy.ts

```diff
@@ -125,16 +139,39 @@ class timeSeriesPolicy {
                         }
                     },
                     message: () => "Time frame must be less than the preceding retention time (" + this.previous().retentionFormatted() + ")"
+                },
+                {
+                    validator: (time: timeValueEntry) => {
+                        const previous = this.previous();
+                        if (!previous || !previous.hasAggregation) {
+                            return true;
+                        }
+
+                        return time.isMultipleOf(previous.aggregation());
+                    },
+                    message: "Time value must be divided by the previous policy aggregation time without a remainder"
                 }
             ]
         });
         
         this.retention.extend({
             validation: [
+                {
+                    validator: (time: timeValueEntry) => !this.hasRetention ||
+                                                         (this.retention().unit() !== 'second' && this.retention().unit() !== 'month') ||
+                                                         time.amount() >= 1,
+                    message: "Minimum value for 'seconds' or 'months' time units is 1",
+                },
                 {
                     validator: (time: timeValueEntry) => !this.hasRetention() || time.isPositive(),
                     message: "Retention time must be greater than zero"
-                }
+                },
+                {
+                    validator: (time: timeValueEntry) => !this.hasRetention ||
+                        (this.retention().unit() !== 'second' && this.retention().unit() !== 'month') ||
+                        time.amount() % 1 === 0,
+                    message: "Please use a whole number for 'seconds' or 'months' time units",
```

Comment:
I wonder what we should write: whole number of integer. ;-)

Reply:
I think 'whole number' 
At least according to https://www.mathsisfun.com/whole-numbers.html